scr/src/main/java/org/apache/felix/scr/impl/manager/DependencyManager.java

```diff
@@ -2239,52 +2238,16 @@ else if ( m_dependencyMetadata.getScope() == ReferenceScope.prototype_required )
         m_componentManager.getLogger().log(LogService.LOG_DEBUG, "Setting target property for dependency {0} to {1}",
                 null, getName(), target );
         BundleContext bundleContext = m_componentManager.getBundleContext();
-        Filter eventFilter = null;
```

Comment:
I tried to find if there is a specified behavior in the DS spec if the target has an invalid filter.  I could not find anything.  Regardless, the current Felix SCR impl seems to log an error here and then create this target `(component.id=-1)` filter to be used instead.

I suggest we still need this check with something like this:
```
        if (target != null)
        {
            try
            {
                FrameworkUtil.createFilter(target);
            }
            catch (InvalidSyntaxException e)
            {
                m_componentManager.getLogger().log(LogService.LOG_ERROR,
                    "Invalid syntax in target property for dependency {0} to {1}", null,
                    getName(), target);

                //create a filter that will never be satisfied  
                target = "(component.id=-1)";
            }
        }
        m_target = target;
```
Where we do the check on the incoming `target` param before setting it to the instance `m_target`

Reply:
I implemented this, but opted to use a different target in case of an invalid one in the component definition, that is guaranteed to be never be satisfied (inspiration drawn from the required capability in the osgi.cmpn API bundle).

Reply:
Good idea, looks good.